File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PagedRequest
    {
        public string CurrentUserName { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 150;

    }
}

File: client/src/app/models/userParams.ts
import { PagedRequest } from './pagination';
import { User } from './users';

export class UserParams implements PagedRequest {
  gender: string;
  minAge = 18;
  maxAge = 99;
  pageNumber = 1;
  pageSize = 6;
  orderBy: 'created' | 'lastActive' = 'lastActive';

  constructor(user: User) {
    this.gender = user.gender === 'male' ? 'female' : 'male';
  }
}
